pkg/services/publicdashboards/database/database.go

```diff
@@ -192,6 +192,17 @@ func (d *PublicDashboardStoreImpl) Update(ctx context.Context, cmd SavePublicDas
 	return err
 }
 
+func (d *PublicDashboardStoreImpl) Delete(ctx context.Context, orgId int64, uid string) error {
+	dashboard := &PublicDashboard{OrgId: orgId, Uid: uid}
+	return d.sqlStore.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
+		_, err := sess.Delete(dashboard)
```

Comment:
```suggestion
		rowsChanged, err := sess.Delete(dashboard)
```

Reply:
Let's change this to return rows changed and push that through the service to the api.

If the api sees that 0 rows have been changed and there is no error, let's return a 404

Reply:
I created another task for this as we talked , in order to finally merge this pr and add some extra tests for this 404 https://github.com/grafana/grafana-partnerships-team/issues/475